File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/descriptor/Descriptor.java
package com.tosmart.xiexuming.libtsparser.descriptor;

import androidx.annotation.NonNull;

import java.util.Arrays;

/**
 * @author xiexuming
 */
public class Descriptor {
    public int descriptorTag;
    public int descriptorLength;
    private byte[] descriptorBuff;

    @NonNull
    @Override
    public String toString() {
        return "Descriptor{" +
                "descriptorTag=" + descriptorTag +
                ", descriptorLength=" + descriptorLength +
                ", descriptorBuff=" + Arrays.toString(descriptorBuff) +
                '}';
    }

    public int getDescriptorTag() {
        return descriptorTag;
    }

    public void setDescriptorTag(int descriptorTag) {
        this.descriptorTag = descriptorTag;
    }

    public int getDescriptorLength() {
        return descriptorLength;
    }

    public void setDescriptorLength(int descriptorLength) {
        this.descriptorLength = descriptorLength;
    }

    public byte[] getDescriptorBuff() {
        return descriptorBuff;
    }

    public void setDescriptorBuff(byte[] descriptorBuff) {
        this.descriptorBuff = descriptorBuff;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/descriptor/Descriptor.kt
package com.example.libtosmartparse.descriptor

import java.util.*

/**
 * @author xxm
 * 2022/7/17
 **/
open class Descriptor {
    private var descriptorTag = 0
    private var descriptorLength = 0
    private lateinit var descriptorBuff: ByteArray

    override fun toString(): String {
        return "Descriptor{" +
                "descriptorTag=" + descriptorTag +
                ", descriptorLength=" + descriptorLength +
                ", descriptorBuff=" + Arrays.toString(descriptorBuff) +
                '}'
    }

    fun getDescriptorTag(): Int {
        return descriptorTag
    }

    fun setDescriptorTag(descriptorTag: Int) {
        this.descriptorTag = descriptorTag
    }

    fun getDescriptorLength(): Int {
        return descriptorLength
    }

    fun setDescriptorLength(descriptorLength: Int) {
        this.descriptorLength = descriptorLength
    }

    fun getDescriptorBuff(): ByteArray {
        return descriptorBuff
    }

    fun setDescriptorBuff(descriptorBuff: ByteArray) {
        this.descriptorBuff = descriptorBuff
    }
}
